ubuntu/python/Dockerfile

```diff
@@ -1,26 +1,30 @@
 FROM databricksruntime/minimal:latest
 
-# Installs python 3.6 and virtualenv for Spark and Notebooks
-RUN apt-get update \
-  && apt-get install -y \
-    python3 \
-    virtualenv \
-  && apt-get clean \
-  && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+RUN apt-get update && apt-get install --yes \
+    wget \
+    libdigest-sha-perl \
+    bzip2
 
-# Initialize the default environment that Spark and notebooks will use
-RUN virtualenv -p python3 --system-site-packages /databricks/python3
+# Download miniconda 4.5.12, then upgrade it to 4.6.12.
+RUN wget --quiet --output-document miniconda.sh https://repo.continuum.io/miniconda/Miniconda3-4.5.12-Linux-x86_64.sh \
+    && (echo '866ae9dff53ad0874e1d1a60b1ad1ef8  miniconda.sh' | md5sum -c) \
+    && (echo 'e5e5b4cd2a918e0e96b395534222773f7241dc59d776db1b9f7fedfcb489157a  miniconda.sh' | shasum -a 256 -c) \
+    # Conda must be installed at /databricks/conda
+    && /bin/bash miniconda.sh -b -p /databricks/conda \
+    && rm miniconda.sh \
+    && /databricks/conda/bin/conda install --name base conda=4.6.12 \
+    # Source conda.sh for all login shells.
+    && ln -s /databricks/conda/etc/profile.d/conda.sh /etc/profile.d/conda.sh
 
-# These python libraries are used by Databricks notebooks and the Python REPL
-# You do not need to install pyspark - it is injected when the cluster is launched
-RUN /databricks/python3/bin/pip install \
-  six==1.12.0 \
-  ipython==7.2.0 \
-  numpy==1.15.4 \
-  pandas==0.23.4 \
-  pyarrow==0.12.0 \
-  matplotlib==3.0.2 \
-  jinja2==2.10
+COPY env.yml /databricks/.conda-env-def/env.yml
 
-# Specifies where Spark will look for the python process
-ENV PYSPARK_PYTHON=/databricks/python3/bin/python3
+RUN /databricks/conda/bin/conda env create --file /databricks/.conda-env-def/env.yml
+
+# Conda recommends using strict channel priority speed up conda operations and reduce package incompatibility problems.
+# Set always_yes to avoid needing -y flags, and improve conda experience in Databricks notebooks.
+RUN /databricks/conda/bin/conda config --system --set channel_priority strict \
+    && /databricks/conda/bin/conda config --system --set always_yes True
+
+# This environment variable must be set to indicate the conda environment to activate.
+ENV DEFAULT_DATABRICKS_ROOT_CONDA_ENV=dcs-minimal
+ENV DATABRICKS_ROOT_CONDA_ENV=dcs-minimal
```

Comment:
Note that currently, we have to set both of these environment variables. The first one is necessary to indicate that this runtime supports conda. The second one is necessary so that the python notebook/repl can be started (won't work without it)